spec/support/sequence_double.rb

```diff
@@ -0,0 +1,12 @@
+# Sequence two arbitrary mock implementations after each other. For example,
+# one can use this to sequence a return after an exception.
+#
+#   helloer = sequenced_double(double("Hello"), -> { raise "yo" }, lambda { 1 })
+#   SayHello.run(helloer)
+def sequenced_double(dbl, *actions)
```

Comment:
I agree, `sequence_double` is a big 'ol hack, it definately needs some love. I like the enumerable solution, I'll give it a whirl and try how it feels